crates/accelerate/src/uc_gate.rs

```diff
@@ -86,11 +122,14 @@ pub fn dec_ucg_help(
     sq_gates: Vec<PyReadonlyArray2<Complex64>>,
     num_qubits: u32,
 ) -> (Vec<PyObject>, PyObject) {
-    let mut single_qubit_gates: Vec<Array2<Complex64>> = sq_gates
+    let mut single_qubit_gates: Vec<Matrix2<Complex64>> = sq_gates
         .into_iter()
-        .map(|x| x.as_array().to_owned())
+        .map(|x| {
+            let res: MatrixView2<Complex64> = x.try_as_matrix().unwrap();
```

Comment:
As discussed offline, we should change this to ``let res: MatrixView2<Complex64, Dyn, Dyn> = x.try_as_matrix().unwrap();``, otherwise the conversion fails.

Reply:
This isn't necessary because the python space change you commented on below is using fortran ordering which matches up with what nalgebra is using. We'd only need dynamic striding like this if we wanted to support an arbitrary 2x2 array. But since this function only is called in one place `uc.py` we don't need that level of flexibility.